app/src/main/java/fr/free/nrw/commons/explore/depictions/DepictsClient.java

```diff
@@ -0,0 +1,186 @@
+import fr.free.nrw.commons.upload.structure.depictions.DepictedItem;
+import fr.free.nrw.commons.utils.CommonsDateUtil;
+import io.reactivex.Observable;
+import io.reactivex.Single;
+
+/**
+ * Depicts Client to handle custom calls to Commons Wikibase APIs
+ */
+@Singleton
+public class DepictsClient {
+
+    private final DepictsInterface depictsInterface;
+    private final MediaInterface mediaInterface;
+    private static final String NO_DEPICTED_IMAGE = "No Image for Depiction";
+
+    @Inject
+    public DepictsClient(DepictsInterface depictsInterface, MediaInterface mediaInterface) {
+        this.depictsInterface = depictsInterface;
+        this.mediaInterface = mediaInterface;
+    }
+
+    /**
+     * Search for depictions using the search item
+     * @return list of depicted items
+     */
+    public Observable<DepictedItem> searchForDepictions(String query, int limit, int offset) {
+        return depictsInterface.searchForDepicts(
+                query,
+                String.valueOf(limit),
+                Locale.getDefault().getLanguage(),
+                Locale.getDefault().getLanguage(),
+                String.valueOf(offset)
+        )
+                .flatMap(depictSearchResponse ->Observable.fromIterable(depictSearchResponse.getSearch()))
+                .map(DepictedItem::new);
+    }
+
+    /**
+     * Get URL for image using image name
+     * Ex: title = Guion Bluford
+     * Url = https://upload.wikimedia.org/wikipedia/commons/thumb/0/04/Guion_Bluford.jpg/70px-Guion_Bluford.jpg
+     */
+    private String getThumbnailUrl(String title) {
+        String baseUrl = "https://upload.wikimedia.org/wikipedia/commons/thumb/";
+        title = title.replace(" ", "_");
+        String MD5Hash =  getMd5(title);
+        /**
+         * We use 70 pixels as the size of our Thumbnail (as it is the perfect fits our UI)
+         */
+        return baseUrl + MD5Hash.charAt(0) + '/' + MD5Hash.charAt(0) + MD5Hash.charAt(1) + '/' + title + "/70px-" + title;
+    }
+
+    /**
+     * Ex: entityId = Q357458
+     * value returned = Elgin Baylor Night program.jpeg
+     */
+    public Single<String> getP18ForItem(String entityId) {
+        return depictsInterface.getImageForEntity(entityId)
+                .map(commonsFilename -> {
+                    String name;
+                    try {
+                        JsonObject claims = commonsFilename.getAsJsonObject("claims").getAsJsonObject();
+                        JsonObject p18 = claims.get("P18").getAsJsonArray().get(0).getAsJsonObject();
+                        JsonObject mainsnak = p18.get("mainsnak").getAsJsonObject();
+                        JsonObject datavalue = mainsnak.get("datavalue").getAsJsonObject();
+                        JsonPrimitive value = datavalue.get("value").getAsJsonPrimitive();
+                        name = value.toString();
+                        name = name.substring(1, name.length() - 1);
+                    } catch (Exception e) {
```

Comment:
Please define concrete classes instead of using `JsonObject`. I guess I have already added some of these classes in my PR #3449. I will base my PR on `structured-data` branch and you could reuse it.

Reply:
This was done as a shortcut originally approved by nicolas If I remember my first investigation of rewriting these to use gson. I think this can be done later. I could find no satisfactory model definitions if I am remembering this corectly